Source/Core/DolphinQt/Config/GameConfigEdit.cpp

```diff
@@ -0,0 +1,169 @@
+      "cause issues. Defaults to <b>True</b>");
+  m_keyword_map[QStringLiteral("FastDiscSpeed")] =
+      tr("Shortens loading times but may break some games. Can have negative effects on "
+         "performance. Defaults to <b>False</b>");
+  m_keyword_map[QStringLiteral("MMU")] = tr("Controls whether or not the Memory Management Unit "
+                                            "should be emulated fully. Few games require it.");
+  m_keyword_map[QStringLiteral("DSPHLE")] =
+      tr("Controls whether to use high or low-level DSP emulation. Defaults to <b>True</b>");
+  m_keyword_map[QStringLiteral("JITFollowBranch")] =
+      tr("Tries to translate branches ahead of time, improving performance in most cases. Defaults "
+         "to <b>True</b>");
+
+  m_keyword_map[QStringLiteral("Gecko")] = tr("Section that contains all Gecko cheat codes.");
+  m_keyword_map[QStringLiteral("ActionReplay")] =
+      tr("Section that contains all Action Replay cheat codes.");
+  m_keyword_map[QStringLiteral("Video_Settings")] =
+      tr("Section that contains all graphics related settings.");
+
+  Connect();
+}
+
+void GameConfigEdit::LoadFile()
+{
+  QFile file(m_path);
+  if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+    return;
+
+  setPlainText(QString::fromStdString(file.readAll().toStdString()));
+}
+
+#include <iostream>
+
+void GameConfigEdit::SaveFile()
+{
+  if (!isVisible() || isReadOnly())
+    return;
+
+  QFile file(m_path);
+
+  if (!file.open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text))
+    return;
+
+  const std::string contents = toPlainText().toStdString();
+
+  file.write(contents.c_str(), contents.length());
+}
+
+void GameConfigEdit::Connect()
+{
+  connect(this, &QTextEdit::textChanged, this, &GameConfigEdit::SaveFile);
+  connect(this, &QTextEdit::selectionChanged, this, &GameConfigEdit::OnSelectionChanged);
+  connect(this, &QTextEdit::customContextMenuRequested, this, &GameConfigEdit::OnContextMenu);
+}
+
+void GameConfigEdit::OnSelectionChanged()
+{
+  const QString& keyword = textCursor().selectedText();
+
+  if (m_keyword_map.count(keyword))
+    QWhatsThis::showText(QCursor::pos(), m_keyword_map[keyword], this);
+}
+
+void GameConfigEdit::AddBoolOption(QMenu* menu, const QString& name, const QString& section,
+                                   const QString& key)
+{
+  auto* option = menu->addMenu(name);
+
+  option->addAction(tr("On"), this,
+                    [this, section, key] { SetOption(section, key, QStringLiteral("True")); });
+  option->addAction(tr("Off"), this,
+                    [this, section, key] { SetOption(section, key, QStringLiteral("False")); });
+}
+
+void GameConfigEdit::SetOption(const QString& section, const QString& key, const QString& value)
+{
+  auto section_cursor = document()->find(QRegExp(QStringLiteral("^\\[%1\\]").arg(section)), 0);
+
+  // Check if the section this belongs in can be found
+  if (!section_cursor.isNull())
```

Comment:
This if and else branch can be inverted so the negation isn't necessary

Reply:
Does it actually matter? The compiler will likely optimize it out, and maybe the inversion helps with reading the control flow here.

Reply:
It's often easier to follow code that performs behavior in terms of the true case when an explicit condition is given, as opposed to one that requires the inverse. Static analysis tools also usually call this sort of thing out for precisely that reason. Which is why I brought it up.

I'm fairly sure @spycrab is capable of answering whether or not it's justified and you don't need to speak for them.

> The compiler...

If the compiler was relevant I would have brought that up and provided an example to back my case up.

Reply:
I wasn't trying to speak for them, only speculating, but fair enough.